src/internal/types.ts

```diff
@@ -198,11 +200,31 @@ export type UpsertProp<T, K extends PropertyKey, V> = Simplify<
         })
 >;
 
-// This type attempts to detect when a type is a single literal value (e.g.
-// "cat"), and not anything else (e.g. "cat" | "dog", string, etc...)
+/**
+ * This type attempts to detect when a type is a single literal value (e.g.
+ * "cat"), and not anything else (e.g. "cat" | "dog", string, etc...).
+ */
 type IsSingleLiteral<K> =
   IsLiteral<K> extends true ? (IsUnion<K> extends true ? false : true) : false;
 
+/**
+ * Widen a literal type.
+ *
+ * @example WidenLiteral<"cat" | "dog"> //=> string
+ * @example WidenLiteral<"cat" | 1> //=> string | number
+ */
+export type WidenLiteral<K> = K extends unknown
```

Comment:
Now that I read the whole code, the name here is confusing because it doesn't just run on literals. It's also a bit limited, if I had a union of objects, wouldn't we expect to widen it just the same? e.g.

```ts
type Narrow = { a: "cat" } | { a: "dog" };

type Widened = WidenLiteral<Narrow>;
//   ^? { a: string };
```